src/converters/toZigbee.ts

```diff
@@ -332,8 +332,20 @@ const converters2 = {
     lock: {
         key: ['state'],
         convertSet: async (entity, key, value, meta) => {
-            utils.assertString(value, key);
-            await entity.command('closuresDoorLock', `${value.toLowerCase()}Door`, {pincodevalue: ''}, utils.getOptions(meta.mapped, entity));
+            // If no pin code is provided, value is a only string. Ex: "UNLOCK"
+            let state = utils.isString(value) ? value.toUpperCase() : null;
+            let pincode = '';
+            // If pin code is provided, value is an object including command and code. Ex: {command: "UNLOCK", code: "1234"}
+            if (utils.isObject(value)) {
+                if (value.code) {
+                    pincode = utils.isString(value.code) ? value.code : '';
+                }
+                if (value.command) {
```

Comment:
```suggestion
                if (value.state) {
```

I think this is more consistent

Reply:
Now I see your suggestion, I fully agree it is more consistent with a key named `state`. I will update that too.

Reply:
Done